src/Catalyst.Modules.UPnP/PortMapper.cs

```diff
@@ -0,0 +1,154 @@
+﻿#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+#endregion
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Mono.Nat;
+using Serilog;
+
+namespace Catalyst.Modules.UPnP
+{
+    public sealed class PortMapper
+    {
+        private readonly SemaphoreSlim _locker = new SemaphoreSlim(1, 1);
+        private readonly INatUtilityProvider _natUtilityProvider;
+        private readonly ILogger _logger;
+
+        public PortMapper(INatUtilityProvider natUtilityProvider, ILogger logger)
+        {
+            _natUtilityProvider = natUtilityProvider;
+            _logger = logger;
+        }
+
+        private async Task<PortMapperConstants.Result> PerformEventOnDeviceFound(int timeoutInSeconds, CancellationToken token, Action<INatDevice> onDeviceFound)
```

Comment:
it is a good practice to have app public methods before a private ones.

Reply:
I've made this change, thanks